packages/roosterjs-editor-api/lib/utils/processList.ts

```diff
@@ -5,65 +5,65 @@ import {
  * So we workaround it by always adding format to list element
  */
-export default function processList(editor: Editor, command: DocumentCommand): Node {
-    if (Browser.isChrome) {
+export default function processList(editor: Editor, command: ValidProcessListDocumentCommands): Node {
+    let currentNode = editor.getElementAtCursor();
+
+    // track the current cursor position with a dummy element.
+    const currentRange = editor.getSelectionRange();
+    if (currentRange) {
+        const selectionNode = fromHtml(SELECTION_NODE_HTML, editor.getDocument())[0];
+        editor.getSelectionRange().insertNode(selectionNode);
+    }
+
+    let clonedNode;
+    if (Browser.isChrome && command == DocumentCommand.Outdent) {
+        // Chrome has some bad behavior when outdenting
+        // in order to work around this, we need to take steps to deep clone the current node
+        // after the outdent, we'll replace the new LI with the cloned content.
+        clonedNode = currentNode.cloneNode(true);
         workaroundForChrome(editor);
     }
 
-    let currentNode = editor.getElementAtCursor();
-    let currentFormat = getComputedStyles(currentNode);
     let existingList = editor.getElementAtCursor('OL,UL');
     editor.getDocument().execCommand(command, false, null);
-    editor.queryElements('.' + TEMP_NODE_CLASS, node => editor.deleteNode(node));
+    let newLIParent: HTMLElement;
+    editor.queryElements('.' + TEMP_NODE_CLASS, node => {
+        newLIParent = node.parentElement;
+        editor.deleteNode(node);
+    });
     let newList = editor.getElementAtCursor('OL,UL');
     if (newList == existingList) {
         newList = null;
     }
 
-    // If this is in a new number list, need to adjust the format of numbers according to its content
-    if (newList && getTagOfNode(newList) == 'OL') {
-        let LIs = ([].slice.call(newList.childNodes) as Node[]).filter(
-            node => getTagOfNode(node) == 'LI'
-        );
-
-        if (LIs.length == 1 && isNodeEmpty(LIs[0], true /*trim*/)) {
-            // When there's only one LI child element which has empty content, it means this LI is just created.
-            // We just format it with current format
-            applyListFormat(LIs[0], currentFormat);
-        } else {
-            // Otherwise, apply the format of first child non-empty element (if any) to LI node
-            for (let li of LIs) {
-                let formatNode = getFirstLeafNode(li);
-                if (formatNode) {
-                    applyListFormat(li, getComputedStyles(formatNode));
-                }
-            }
+    if (newList && clonedNode && newLIParent) {
+        // if the clonedNode and the newLIParent share the same tag name
+        // we can 1:1 swap them
+        if ((clonedNode instanceof HTMLElement) && clonedNode.tagName == newLIParent.tagName) {
+            newList.replaceChild(clonedNode, newLIParent);

```

Comment:
Can we preserve all child nodes under the LI nodes rather than use cloned nodes? Because plugin may hold reference to node under LI.

Reply:
Unfortunately, I'm not sure we can. Looking at how chrome behaves, it seems that the node references get broken when re-parenting, even on master. Maybe I'm missing something, but this scenario may be broken currently, not just by the new change I'm proposing.